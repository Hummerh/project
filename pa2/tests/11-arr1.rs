fn main() {
      let a : [bool; 3] = [true, false, false];
      let x = a[0];
      a[0];
      a[1 - 1];
      a[if (a[0]) { 0 } else { 1 }];
}
