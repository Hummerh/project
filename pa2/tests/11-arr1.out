(crate
    (items
        (fn-def
            (id
                (main))
            (block
                (let
                    (pat-id
                        (id
                            (a)))
                    (type-arr
                        (type-bool)
                        (lit-dec))
                    (arr
                        (exprs
                            (true)
                            (false)
                            (false))))
                (let
                    (pat-id
                        (id
                            (x)))
                    (arr-index
                        (id
                            (a))
                        (lit-dec)))
                (arr-index
                    (id
                        (a))
                    (lit-dec))
                (arr-index
                    (id
                        (a))
                    (sub
                        (lit-dec)
                        (lit-dec)))
                (arr-index
                    (id
                        (a))
                    (if
                        (arr-index
                            (id
                                (a))
                            (lit-dec))
                        (block
                            (lit-dec))
                        (block
                            (lit-dec))))))))